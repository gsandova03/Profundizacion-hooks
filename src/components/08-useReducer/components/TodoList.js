import React from 'react';
import PropTypes from 'prop-types';
import TodoListItem from './TodoListItem';

const TodoList = ({ handleComplete, handleDelete, todos }) => {
  return (
    <>
     <ul className="list-group list-group-flush">
        {
          todos.map( ( todo,i ) => (
            <TodoListItem 
              key={ todo.id } 
              id={ todo.id } 
              index={ i } 
              todo={ todo } 
              handleComplete={ handleComplete } 
              handleDelete={ handleDelete } 
            />
          ))
        }
      </ul> 
    </>
  )
}

TodoList.propTypes = {
  handleComplete: PropTypes.func.isRequired,
  handleDelete: PropTypes.func.isRequired,
  todos: PropTypes.array
}

export default TodoList;
