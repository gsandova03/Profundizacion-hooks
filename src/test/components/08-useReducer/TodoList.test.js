import React from 'react'
import { shallow } from 'enzyme/build'
import TodoList from "../../../components/08-useReducer/components/TodoList"
import { demoTodo } from '../../fixtures/demoTodo'



describe('Pruebas en el componente <TodoList/>', () => {

  const handleComplete = jest.fn()
  const handleDelete = jest.fn()

  const wrapper = shallow( 
    <TodoList 
      todos={ demoTodo }
      handleComplete={ handleComplete }
      handleDelete={ handleDelete }
    /> 
  )
  
  test('debe de mostrarse correctamente', () => {
    expect( wrapper ).toMatchSnapshot()
  })

  test('debe de tener 2 <TodoListItem/>', () => {
    expect( wrapper.find('TodoListItem').length ).toBe( demoTodo.length )
    expect( wrapper.find('TodoListItem').at(0).prop('handleDelete') ).toEqual( expect.any(Function) );
  })
  
})
