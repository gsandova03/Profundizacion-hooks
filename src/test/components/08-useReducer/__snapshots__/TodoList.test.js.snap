// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <TodoList/> debe de mostrarse correctamente 1`] = `
<Fragment>
  <ul
    className="list-group list-group-flush"
  >
    <TodoListItem
      handleComplete={[MockFunction]}
      handleDelete={[MockFunction]}
      id={1}
      index={0}
      key="1"
      todo={
        Object {
          "done": false,
          "id": 1,
          "todo": "aprender react",
        }
      }
    />
    <TodoListItem
      handleComplete={[MockFunction]}
      handleDelete={[MockFunction]}
      id={2}
      index={1}
      key="2"
      todo={
        Object {
          "done": false,
          "id": 2,
          "todo": "aprender mongo",
        }
      }
    />
  </ul>
</Fragment>
`;
